### PR TITLE
Road to v2

### PR DESCRIPTION
**Pull request by SecretBase**
- [x] using yarn
- [x] Integrate eslint
- [x] Fix all lint errors
- [x] Refactor internal implementation and pass all test. Make it more declarative
- [x] Add build pipeline for node modules and web modules (umd)
- [x] Make example code for browser usage 
- [x] Update README for browser support and ployfill guide

This is major version bump for refactoring with backward compatibility.

it suppose to be no migration from v1 to v2.

@adamliuxy please have a look.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,11 +23,22 @@
...
+		"babel-cli": "^6.26.0",
+		"babel-loader": "^7.1.2",
+		"babel-minify-webpack-plugin": "^0.2.0",
+		"babel-polyfill": "^6.26.0",
+		"babel-preset-env": "^1.6.0",
+		"eslint": "^4.5.0",
+		"eslint-config-aftership": "^4.0.2",
+		"eslint-plugin-import": "^2.7.0",
+		"jest": "^21.2.1",
+		"should": "latest",
```
Comment (hans-lizihan) [inline package.json]: you dont need this.

**Comment by hans-lizihan** [comment]
1. how do you publish this pacakge? `aftership-publish`? or pure `npm publish`
2. how to build? when to build? prepublish hook?

File: lib/index.js [hunk 2]
```diff
@@ -1,169 +1,78 @@
...
-		} else {
-			// B: no country, no plus sign --> treat it as USA
-			// 1. check length if == 11, or 10, if 10, add +1, then go go D
-			// no plus sign, no country is given. then it must be USA
-			// iso3166 = iso3166_data[0]; already assign by the default value
-			if (iso3166.phone_number_lengths.indexOf(phone.length) !== -1) {
-				phone = '1' + phone;
-			}
 		}
+	} else if (plus_sign) {
```
Comment (hans-lizihan) [inline lib/index.js]: i think the logic here did not preserve the logic on the left.

if `formatted country` is falsy and `plus_sign` is truth then the behavior is different.

**Comment by SecretBase** [comment]
I am going to merge it and publish a release candidate version.

**Pull request merged**